src/components/UserSelector.tsx

```diff
@@ -1,35 +1,84 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import classNames from 'classnames';
+import { User } from '../types/User';
+import { getUsers } from './api/users';
+
+type Props = {
+  onSelect: (user: User) => void,
+  selectedUser: User | null,
+};
+
+export const UserSelector: React.FC<Props> = ({
+  onSelect,
+  selectedUser,
+}) => {
+  const [users, setUsers] = useState<User[] | null>(null);
+  const [showList, setShowList] = useState(false);
+
+  useEffect(() => {
+    getUsers().then(setUsers);
+
+    return () => setUsers(null);
+  }, []);
+
+  const handleUserSelect = (user: User) => {
+    onSelect(user);
+    setShowList(false);
+  };
 
-export const UserSelector: React.FC = () => {
   return (
     <div
       data-cy="UserSelector"
-      className="dropdown is-active"
+      className={classNames(
+        'dropdown',
+        { 'is-active': showList },
+      )}
     >
       <div className="dropdown-trigger">
         <button
           type="button"
           className="button"
           aria-haspopup="true"
           aria-controls="dropdown-menu"
+          onClick={() => {
+            setShowList(true);
```

Comment:
You can use a shorter syntax for the onClick handler: `onClick={() => setShowList(true)}`.